src/pypdfium2/_helpers/textinserter.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: 2022 Anurag Bansal <anurag.bansal.585@gmail.com>
+# SPDX-FileCopyrightText: 2022 geisserml <geisserml@gmail.com>
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+import ctypes
+from pypdfium2 import _pypdfium as pdfium
+from pypdfium2._helpers.opener import open_page
+
+try:
+    import uharfbuzz as harfbuzz
+except ImportError:
+    have_harfbuzz = False
+else:
+    have_harfbuzz = True
+
+
+def insert_text(
+        pdf,
+        page_index,
+        text,
+        pos_x,
+        pos_y,
+        font_path,
+        font_size,
+        font_type,
+        font_is_cid,
+    ):
+    """
+    Insert text into a PDF page at a specified position. This function supports Asian scripts such as Hindi.
+    There is no position validation, so make sure the text will be within page boundaries.
+    ``uharfbuzz`` is required as an additional dependency.
+    
+    Parameters:
+        pdf (``FPDF_DOCUMENT``):
+            PDFium document handle.
+        page_index (int):
+            Zero-based index of the page to edit.
+        text (str):
+            The message to insert.
+        pos_x (float):
+            Distance from left border to first character.
+        pos_y (float):
+            Distance from bottom border to first character.
+        font_path (str):
+            File path of the font to use.
+        font_size (float):
+            Font size to use.
+        font_type (int):
+            State the type of the given font (``FPDF_FONT_TYPE1`` or ``FPDF_FONT_TRUETYPE``).
+        font_is_cid (bool):
+            State whether the given font is a CID font.
+    """
+    
+    if not have_harfbuzz:
+        raise RuntimeError("Function insert_text() requires uharfbuzz to be installed.")
+    
+    with open(font_path, "rb") as fh:
+        font_data = fh.read()
+    
+    page = open_page(pdf, page_index)
+    
+    hb_blob = harfbuzz.Blob.from_file_path(font_path)
+    hb_face = harfbuzz.Face(hb_blob)
+    hb_font = harfbuzz.Font(hb_face)
+    hb_fontscale = hb_font.scale[0]
+    
+    pdf_font = pdfium.FPDFText_LoadFont(
+        pdf,
+        ctypes.cast(font_data, ctypes.POINTER(ctypes.c_uint8)),
+        len(font_data),
+        font_type,
+        font_is_cid,
+    )
+    
+    hb_buffer = harfbuzz.Buffer()
```

Comment:
I agree that this is the logic I provided. Here I mentioned to create the hb_buffer object for each text typesetting info we add. Honestly, it was a bit of lazy code writing on my end, and there should probably be something better available using which the buffer object is not required to be created again and again.

Reply:
I'm not exactly sure what you mean? Do you think the buffer object should be initialised in a different scope?

Reply:
I meant as of now, we are creating the buffer each and every time the create text function is loaded. And the solution does make sense considering that when we try to fetch the glyph data for more text which can be inserted into the hb buffer, the solution does not work any more and returns empty blocks (you can try this behaviour).

Now if hb_buffer creation in itself is not a heavy task, then I don't think that any update is required, but if it is something which can create problems in large PDFs then this probably needs some investigation.

Obviously, I am using this version on my end, as I do not have a better alternative for this on my end.

Reply:
OK, thanks for the clarification.